docs/fastdds/getting_started/simple_app/includes/subscriber.rst

```diff
@@ -0,0 +1,122 @@
+    Is the object responsible for the actual reception of the data.
+    It registers in the application the topic (TopicDescription) that identifies the data to be read and
+    accesses the data received by the subscriber.
+*   :class:`DataReaderListener`.
+    This is the listener assigned to the data reader.
+*   :class:`DataReaderQoS`.
+    Structure that defines the QoS of the DataReader.
+*   :class:`SampleInfo`.
+    Is the information that accompanies each sample that is ‘read’ or ‘taken.’
+
+.. literalinclude:: /../code/Examples/C++/DDSHelloWorld/src/HelloWorldSubscriber.cpp
+    :language: C++
+    :lines: 22,29
+
+The next line defines the :class:`HelloWorldSubscriber` class that implements a subscriber.
+
+.. literalinclude:: /../code/Examples/C++/DDSHelloWorld/src/HelloWorldSubscriber.cpp
+    :language: C++
+    :lines: 33
+
+Starting with the private data members of the class, it is worth mentioning the implementation of the data reader
+listener.
+The private data members of the class will be the participant, the subscriber, the topic, the data reader, and the
+data type.
+As it was the case with the data writer, the listener implements the callbacks to be executed in case an event
+occurs.
+The first overridden callback of the SubListener is the ``on_subscrition_matched``, which is the analog of the
+``on_publication_matched`` callback of the DataWriter.
+
+.. literalinclude:: /../code/Examples/C++/DDSHelloWorld/src/HelloWorldSubscriber.cpp
+    :language: C++
+    :lines: 60-77
+
+The second overridden callback is ``on_data_available``.
+In this, the next received sample that the data reader can access is taken and processed to display its content.
+It is here that the object of the :class:`SampleInfo` class is defined, which determines whether a sample has already
+been read or taken.
+Each time a sample is read, the counter of samples received is increased.
+
+.. literalinclude:: /../code/Examples/C++/DDSHelloWorld/src/HelloWorldSubscriber.cpp
+    :language: C++
+    :lines: 79-92
+
+The public constructor and destructor of the class is defined below.
+
+.. literalinclude:: /../code/Examples/C++/DDSHelloWorld/src/HelloWorldSubscriber.cpp
+    :language: C++
+    :lines: 102-126
+
+Then we have the subscriber initialization public member function.
+This is the same as the initialization public member function defined for the :class:`HelloWorldPublisher`.
+The QoS configuration for all entities, except for the participant's name, is the default QoS
+(``PARTICIPANT_QOS_DEFAULT``, ``SUBSCRIBER_QOS_DEFAULT``, ``TOPIC_QOS_DEFAULT``, ``DATAREADER_QOS_DEFAULT``).
+The default value of the QoS of each DDS Entity can be checked in the
+`DDS standard <https://www.omg.org/spec/DDS/About-DDS/>`_.
+
+.. literalinclude:: /../code/Examples/C++/DDSHelloWorld/src/HelloWorldSubscriber.cpp
+    :language: C++
+    :lines: 128-168
+
+The public member function ``run()`` ensures that the subscriber runs until all the samples have been received.
```

Comment:
I'd write something saying that this is an active wait.

Reply:
In fact, we could make a small sleep inside while loop (1ms or so) to ease the CPU